pkg/river/internal/value/value.go

```diff
@@ -106,6 +106,12 @@ func Encode(v interface{}) Value {
 	return makeValue(reflect.ValueOf(v))
 }
 
+// FromRaw converts a reflect.Value into a River Value. It is useful to prevent
+// downcasting a interface into an any.
+func FromRaw(v reflect.Value) Value {
+	return makeValue(v)
+}
```

Comment:
The need to do this is probably the start of a wider discussion around whether interface types should be capsules. It's fairly hard to reason about in the internal code, and there's still some edge cases where a capsule can be downcasted like when providing it to a `vm.Scope`. 

While it might be more work on component developers, it might be better to easier to reason about not doing things this way. 

cc @grafana/databases-agent

Reply:
When you say marker interface, do you mean interfaces would need to have the `Capsule interface` on them?

Reply:
Yeah, I mean to change the rules so that:

1. Go interfaces are deferenced into their underlying type. 
2. Types which implement `river.Capsule` are forced capsule types. 
3. Go types which have no River mapping (like channels) default to capsule types.

The only new rule here is the first one; currently interface types with at least one method are treated as capsules, though it's too easy for that information to be lost.

Reply:
Either way, I think making these changes should be something we do later, we still need this bug fix for now, especially since #2431 probably makes it happen for even more components.